solutions/swb-app/src/projectRoutes.ts

```diff
@@ -76,26 +83,44 @@ export function setUpProjectRoutes(
       async function checkDependencies(projectId: string): Promise<void> {
         // environments
         const projectHasEnvironments = environmentService.doesDependencyHaveEnvironments(projectId);
+
         // datasets
-        const projectHasDatasets = projectService.checkDependency('dataset', projectId);
+        const datasetDependency = metadataService.listDependentMetadata<ProjectDatasetMetadata>(
+          resourceTypeToKey.project,
+          projectId,
+          resourceTypeToKey.dataset,
+          ProjectDatasetMetadataParser,
+          { pageSize: 1 }
+        );
+
         // etcs
-        const projectHasEnvTypeConfigs = projectService.checkDependency('envTypeConfig', projectId);
-        const [envBoolean, datasetBoolean, envTypeConfigBoolean] = await Promise.all([
+        const envTypeConfigDepedency = metadataService.listDependentMetadata<ProjectEnvTypeConfigMetadata>(
+          resourceTypeToKey.project,
+          projectId,
+          resourceTypeToKey.envTypeConfig,
+          ProjectEnvTypeConfigMetadataParser,
+          { pageSize: 1 }
+        );
+
+        const [envBoolean, { data: datasets }, { data: envTypeConfigs }] = await Promise.all([
```

Comment:
Wonder whether we want to rename `envBoolean` -> `hasEnvironments`

Reply:
I'm good with that. Let me push change really quick.